### ISSUE TITLE
dex: handle zero-output in `SwapExecution::max_price` 

### ISSUE THREAD

**Issue by erwanor** [penumbra-zone/penumbra#3893]
Every block, swaps are batched together and piped to the DEX engine. We get back a `SwapExecution` which is an execution trace, recording quantities of asset routed over the trading pair as well as aggregate value.

```rust
pub struct SwapExecution {
    pub traces: Vec<Vec<Value>>,
    pub input: Value,
    pub output: Value,
}
```

As part of `route_and_fill` we alternate spill/fill phases, deciding whether or not to continue based on the price recorded by the swap execution:

```rust
    let Some(accurate_max_price) = execution.max_price()? else {
        tracing::debug!("no traces in execution, exiting route_and_fill");
        break;
    };
```

To get the "max price" from a trace of swaps, we lookup the last one (the traces are ordered since better positions get matched first) and compute `input/output`.

However, this has an obvious failure mode: what if the output is zero? Recall that we compute the output $\Lambda_2 = \lfloor {\frac {p_1} {p_2} \Delta_1} \rfloor$, so for a position with a sufficiently small effective price, we will have $\Lambda_2$ round down to zero. In this case, the `max_price` computation will fail because dividing by zero is not allowed.

**Comment by hdevalence** [comment]
What's the right way to handle a "dust trace" like this?

**Comment by erwanor** [comment]
Context for this comment: Testnet 67.1 halted, hitting this incorrect assumption.

We need to approach a fix cautiously, the primary consumer for the `route_and_fill` price limit branch is the arbitrage logic loop.

Observations:

I think there are couple different things wrong that all resolve into this incorrect assumption that the `route_and_fill` logic makes about output being nonzero. My current working hypothesis for the specific chain reaction causing the halt is that during arbitrage, speculatively executing the selected frontier resulted in an `actual_price` that was above the `spill_price`. As a result, the frontier transaction was not applied, and the execution ended with `output = 0`.

**Comment by hdevalence** [comment]
Closed by #3949
